calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -162,15 +162,26 @@ func (rh resourceHelper) Apply(ctx context.Context, client client.Interface, res
 	// Try to create the resource first.
 	ro, err := rh.Create(ctx, client, resource)
 
-	// Fall back to an Update if the resource already exists, or the datastore does not support
-	// create operations for that resource.
 	switch err.(type) {
-	case cerrors.ErrorResourceAlreadyExists, cerrors.ErrorOperationNotSupported:
+	case cerrors.ErrorResourceAlreadyExists:
 		// Insert the original ResourceVersion back into the object before trying the Update.
 		resource.(ResourceObject).GetObjectMeta().SetResourceVersion(originalRV)
 
-		// Try updating if the resource already exists.
+		// Fall back to an Update if the resource already exists.
 		return rh.Update(ctx, client, resource)
+
+	case cerrors.ErrorOperationNotSupported:
+		// Insert the original ResourceVersion back into the object before trying the Update.
+		resource.(ResourceObject).GetObjectMeta().SetResourceVersion(originalRV)
+
+		// Try updating if the resource already exists.
+		_, nerr := rh.Update(ctx, client, resource)
+		if _, dne := nerr.(cerrors.ErrorResourceDoesNotExist); dne {
```

Comment:
it's more common to use `ok` to indicate successful casting (rather than using `dne`